src/GenerateUnionAttribute/UnionAttributeOptions.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Dunet.UnionAttributeGeneration;
+
+record UnionAttributeOptions(bool GenerateFactoryMethods, string FactoryMethodPrefix, string FactoryMethodSuffix)
+{
+    public static readonly UnionAttributeOptions Default = new(true, "New", "");
```

Comment:
Would it be better to have factory method generation be opt-in rather than enabled by default?

Reply:
Agreed. I'm changing this PR to have project-level defaults that can then be overridden with the properties on `UnionAttribute`